chain/db/statedb.go

```diff
@@ -263,3 +271,37 @@ func (sdb *StateDB) SetID(addr common.Uint160, id []byte) error {
 	account.SetID(id)
 	return nil
 }
+
+func (sdb *StateDB) UpdateBalance(addr common.Uint160, value common.Fixed64, op Operation) error {
+	acc := sdb.GetOrNewAccount(addr)
+	amount := acc.GetBalance()
+
+	switch op {
+	case Addition:
+		acc.SetBalance(amount + value)
+	case Subtraction:
+		if amount < value {
+			return errors.New("UpdateBalance: no sufficient funds")
+		}
+		acc.SetBalance(amount - value)
+	default:
+		return errors.New("UpdateBalance: invalid operation")
+	}
+
+	return nil
+}
+
+func (sdb *StateDB) IncrNonce(addr common.Uint160) error {
```

Comment:
Can we guarantee that IncrNonce will not be called concurrently?

Reply:
account is proccessed sequentially now.